### PR TITLE
Update 'SearchOptionsRunTypes' for  'lastModifiedDateRange' 

### PR DESCRIPTION
**Pull request by PenghaiZhang**
#2623 

##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

At the moment,  the Runtypes used for 'lastModifiedDateRange'  is `Record`, which means if either the start or end of a date range is undefined the Runtypes guard for `DehydratedSearchOptionsRunTypes` will fail. As a result, the feature of shared search does not work.

So in this PR I change the Runtypes used for 'lastModifiedDateRange' to `Partial`.

### CODE REVIEW

**Review by edalex-ian — COMMENTED** [review]
Looks good @PenghaiZhang . But... Seeing this was to address a bug that was only found through manual testing, does that mean we're missing a test? Or is it too hard to write a test for?

**Comment by PenghaiZhang** [comment]
> Looks good @PenghaiZhang . But... Seeing this was to address a bug that was only found through manual testing, does that mean we're missing a test? Or is it too hard to write a test for?

I think we have one test for this feature but the test is too simple. I will try to rewrite the test.

File: Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts [hunk 1]
```diff
@@ -144,6 +147,65 @@ describe("SearchModule", () => {
       expect(convertedParamsPromise).toEqual(allSearchOptions);
     });
 
+    const emptyStartDateQueryString =
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts:150]: Maybe I'm confused, but shouldn't this be called `emptyEndDateQueryString` and the one below `emptyStartDateQueryString`?

File: Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts [hunk 2]
```diff
@@ -144,6 +147,65 @@ describe("SearchModule", () => {
...
+      '{"lastModifiedDateRange":{"start":"2020-05-26T03:24:00.889Z"}}';
+    const expectedEmptyStartDateSearchOptions: SearchOptions = {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts:152]: Similar to above comment.

**Review by edalex-ian — COMMENTED** [review]
Ummm, looks good but did you get the variable names around the wrong way? Or have I misread?

File: Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts (hunk 1 above)
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/modules/SearchModule.test.ts:150]: My bad. I will change the names.

**Review by edalex-ian — APPROVED** [review]

**Pull request merged**